parquet-column/src/main/java/org/apache/parquet/io/api/Binary.java

```diff
@@ -48,6 +50,14 @@ private Binary() { }
 
   abstract public byte[] getBytes();
 
+  /**
+   * Variant of getBytes() that avoids copying backing data structure
+   * @return backing byte[] if possible, else returns result of getBytes()
+   */
+  abstract public byte[] getBytesUnsafe();
```

Comment:
maybe add to the docs that this will always be a byte[] of the correct size, with an offset of 0. It just might be a reference to a byte array that you aren't supposed to hold on to or modify unless you know what you're doing. right?